misc/python/materialize/cli/crate_diagram.py

```diff
@@ -107,7 +107,7 @@ def main(show: bool, diagram_file: Optional[str], roots: List[str]) -> None:
     if roots:
         (local_deps, areas) = filter_to_roots(areas, local_deps, roots)
 
-    diagram_file = MZ_ROOT / diagram_file
+    diagram_file = str(MZ_ROOT / diagram_file)
```

Comment:
Why do we need to change the type from `Path` to `str`?
Nit if we keep it like this: casting in 114 will no longer be necessary.